app/controllers/katello/api/rhsm/candlepin_proxies_controller.rb

```diff
@@ -381,13 +381,7 @@ def get_content_view_environments(label = nil, organization = nil)
           where("#{Organization.table_name}.id = ?", organization.id)
       environments = environments.where("#{Katello::ContentViewEnvironment.table_name}.label = ?", label) if label
 
-      environments.delete_if do |env|
-        if env.content_view.default
-          !env.environment.readable?
-        else
-          !env.content_view.readable?
-        end
-      end
+      environments = environments.reject { |env| (env.content_view.default && !env.environment.readable?) || !env.content_view.readable? }
```

Comment:
@johnpmitsch Ok, but ruby still supports it. http://ruby-doc.org/core-2.2.0/Array.html#method-i-delete_if

"Convert to an Array by calling #to_a before using these methods."